From 7d571983f5794f88c92560f1fc5061aa4ac12bac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Diego=20Elio=20'Flameeyes'=20Petten=C3=B2?= <flameeyes@gmail.com>
Date: Mon, 25 Jan 2010 19:09:53 +0100
Subject: [PATCH] Don't fail away if launchy could not be loaded.

Since it's just a minor feature, make launchy optional this way.
---
 bin/shotgun |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/bin/shotgun b/bin/shotgun
index d618d29..e12c4d8 100755
--- a/bin/shotgun
+++ b/bin/shotgun
@@ -137,7 +137,11 @@ server.run app, options do |inst|
   puts "== Shotgun starting #{server.to_s} on http://#{options[:Host]}:#{options[:Port]}"
 
   if browse
-    require 'launchy'
-    Launchy.open("http://#{options[:Host]}:#{options[:Port]}#{options[:Path]}")
+    begin
+      require 'launchy'
+      Launchy.open("http://#{options[:Host]}:#{options[:Port]}#{options[:Path]}")
+    rescue LoadError
+      $stderr.puts "Unable to load 'launchy', the browser won't be launched."
+    end
   end
 end
-- 
1.6.6.1

